### PR TITLE
fix: increase linter deadline to 15 minutes

### PR DESCRIPTION
**Pull request by ccojocar**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: hack/linter.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ golangci-lint run \
     -E gosec \
     -E gofmt \
     --skip-dirs vendor \
-    --deadline 5m0s \
+    --deadline 15m0s \
```
Comment (hferentschik) [inline hack/linter.sh]: 15min !? Does linting take so long?

Comment (ccojocar) [reply hack/linter.sh]: It's about 5 min right now. And it seems to timeout occasionally resulting in a build failure.

**Comment by abayer** [comment]
/lgtm

Seems reasonable. We’ll want to watch to see if timeouts still happen with the increased duration, of course - that would be a sign of a legit hanging problem that increasing the timeout further would be unlikely to fix.

**Pull request merged**